src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -143,6 +239,17 @@ void AggregatedDataVariants::convertToTwoLevel()
     }
 }
 
+void AggregatedDataVariants::serialize(WriteBuffer & wb, const Aggregator & aggregator_) const
+{
+    aggregator_.checkpoint(*this, wb);
```

Comment:
AggregatedDataVariants has an aggregator point, what's the difference between the aggregator passed in and the one it points to ?

Reply:
if there is no data, the member variable `aggregator`  in AggregatedDataVariants is nullptr, so we can not use itself aggregator.

Reply:
commented here

Reply:
No, it's nullptr unless we explicitly assign it before